CHANGELOG.md

```diff
@@ -1,5 +1,125 @@
+
+    There are a number of requirements around _what_ your Workers script is named – previously, Wrangler would opaquely fail and not indicate that your script name was invalid: this PR updates the `init` and `generate` commands to validate the potential name before continuing to create a new project.
+
+    [gabbifish]: https://github.com/gabbifish
+    [issue/470]: https://github.com/cloudflare/wrangler/issues/470
+    [pull/686]: https://github.com/cloudflare/wrangler/pull/686
+
+  - **Ensure KV subcommands check for presence of required fields in wrangler.toml - [gabbifish], [issue/607] [pull/665]**
+
+    There's a number of commands in `wrangler` that require a properly configured `wrangler.toml` file: instead of failing, this PR ensures that these commands now check your configuration file before attempting any action. Hooray for clarity! 😇
+
+    [gabbifish]: https://github.com/gabbifish
+    [pull/665]: https://github.com/cloudflare/wrangler/pull/665
+
+  - **Show size when a KV upload error occurs - [stevenfranks], [issue/650] [pull/651]**
+
+    Previously, when uploading a file to Workers KV from Wrangler, the error output didn't indicate the size of the file that was being uploaded. This PR improves the output of that message by showing both the file size and the maximum file size that can be uploaded to Workers KV.
+
+    [stevenfranks]: https://github.com/stevenfranks
+    [issue/650]: https://github.com/cloudflare/wrangler/issues/650
+    [pull/651]: https://github.com/cloudflare/wrangler/pull/651
+
+  - **Better file support for kv:put - [phayes], [pull/633]**
+
+    The `kv:put` command, introduced in Wrangler 1.3.1, has been improved to support uploading non-UT8 files. In addition, the command now streams files directly to the Cloudflare API when uploading, instead of buffering them in memory during the upload process.
+
+    [phayes]: https://github.com/phayes
+    [pull/633]: https://github.com/cloudflare/wrangler/pull/633
+
+- ### Fixes
+
+  **Ensure we install and cache the latest version of cargo-generate and wasm-pack if user has an outdated cargo installed version - [EverlastingBugstopper], [issue/666] [pull/726]**
+
+    Wrangler orchestrates a few other tools under the hood, notably [`wasm-pack`] and [`cargo-generate`]. We use a library called [`binary-install`] to fetch and cache binaries we download. However, to avoid downloading unecessarily, we first check if the user has a copy locally on their machine that they had `cargo install`'d. We had a bug where in this logic branch, we *didn't* check that the local version was the most up-to-date version. This meant that users who had an older installed version may run into errors when wrangler expected to use features of a newer version of that tool. This PR adds the logic to check for the version and will install and cache a newer version for wrangler to use (leaving your local version as is!).
```

Comment:
just curious, why square brackets around package names?

Reply:
ah! i think ashley meant for these to be markdown links - i'll add